deepstream/5.1/deepstream_ssd_parser.py

```diff
@@ -0,0 +1,518 @@
+    # Here's an example of inserting a single object.
+    obj_meta = pyds.nvds_acquire_obj_meta_from_pool(batch_meta)
+    # Set bbox properties. These are in input resolution.
+    rect_params = obj_meta.rect_params
+    rect_params.left = int(IMAGE_WIDTH * frame_object.left)
+    rect_params.top = int(IMAGE_HEIGHT * frame_object.top)
+    rect_params.width = int(IMAGE_WIDTH * frame_object.width)
+    rect_params.height = int(IMAGE_HEIGHT * frame_object.height)
+
+    # Semi-transparent yellow backgroud
+    rect_params.has_bg_color = 0
+    rect_params.bg_color.set(1, 1, 0, 0.4)
+
+    # Red border of width 3
+    rect_params.border_width = 3
+    rect_params.border_color.set(1, 0, 0, 1)
+
+    # Set object info including class, detection confidence, etc.
+    obj_meta.confidence = frame_object.detectionConfidence
+    obj_meta.class_id = frame_object.classId
+
+    # There is no tracking ID upon detection. The tracker will
+    # assign an ID.
+    obj_meta.object_id = UNTRACKED_OBJECT_ID
+
+    lbl_id = frame_object.classId
+    if lbl_id >= len(label_names):
+        lbl_id = 0
+
+    # Set the object classification label.
+    obj_meta.obj_label = label_names[lbl_id]
+
+    # Set display text for the object.
+    txt_params = obj_meta.text_params
+    if txt_params.display_text:
+        pyds.free_buffer(txt_params.display_text)
+
+    txt_params.x_offset = int(rect_params.left)
+    txt_params.y_offset = max(0, int(rect_params.top) - 10)
+    txt_params.display_text = (
+        label_names[lbl_id] + " " + "{:04.3f}".format(frame_object.detectionConfidence)
+    )
+    # Font , font-color and font-size
+    txt_params.font_params.font_name = "Serif"
+    txt_params.font_params.font_size = 10
+    # set(red, green, blue, alpha); set to White
+    txt_params.font_params.font_color.set(1.0, 1.0, 1.0, 1.0)
+
+    # Text background color
+    txt_params.set_bg_clr = 1
+    # set(red, green, blue, alpha); set to Black
+    txt_params.text_bg_clr.set(0.0, 0.0, 0.0, 1.0)
+
+    # Inser the object into current frame meta
+    # This object has no parent
+    pyds.nvds_add_obj_meta_to_frame(frame_meta, obj_meta, None)
+
+
+def pgie_src_pad_buffer_probe(pad, info, u_data):
+
+    gst_buffer = info.get_buffer()
+    if not gst_buffer:
+        print("Unable to get GstBuffer ")
+        return
+
+    # Retrieve batch metadata from the gst_buffer
+    # Note that pyds.gst_buffer_get_nvds_batch_meta() expects the
+    # C address of gst_buffer as input, which is obtained with hash(gst_buffer)
+    batch_meta = pyds.gst_buffer_get_nvds_batch_meta(hash(gst_buffer))
+    l_frame = batch_meta.frame_meta_list
+
+    detection_params = DetectionParam(CLASS_NB, ACCURACY_ALL_CLASS)
+    box_size_param = BoxSizeParam(IMAGE_HEIGHT, IMAGE_WIDTH,
+                                  MIN_BOX_WIDTH, MIN_BOX_HEIGHT)
+    nms_param = NmsParam(TOP_K, IOU_THRESHOLD)
+
+    label_names = get_label_names_from_file("ssd_mobilenet_v2/labels.txt")
```

Comment:
Please don't use hard coded path. Put the input argument instead even if this is a constant address.

Reply:
modified, thanks.